tests/jballerina-unit-test/src/test/resources/test-src/types/function/function-type-negative.bal

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+function bar(int b) returns int {
+    return b;
+}
+
+isolated function bar2(int b) returns int {
+    return b;
+}
+
+function baz = bar;
+
+isolated function baz2 = bar2;
+
+function functionWithFunctionReturnType() returns function {
+    string a = "hello";
+    return a;
+}
+
+function testFunctionTypedescWithIsolatedQualifier() {
+    isolated function func = bar;
+}
+
+function testFunctionTypedescWithTransactionalQualifier() {
+    transactional function func;
+}
+
+function castFunctionVariable() {
+    function func = bar;
+    function(int) returns string val= <function(int) returns string> func;
+}
```

Comment:
Nothing is being asserted to be negative here, right?

Reply:
Removed.